11-Backbonejs/Backbone-Models.md

```diff
@@ -0,0 +1,326 @@
+
+```javascript
+// src/app.js
+
+...
+ 
+   // prints "Title:  Create a model"
+console.log("Title: " + my_task.get("title") );
+  // prints "Complete:  true"
+console.log("Complete: " + my_task.get("completed") );
+```
+
+The `set` method allows you to change the attributes of a Model by either passing it a key and value or by passing `set` a JSON object.
+
+```javascript
+// sample code illustrating `set`
+
+...
+
+  // key & value example
+my_task.set("title", "Test the `set` method.");
+  // prints "Title:  Test the `set` method."
+console.log("Title: " + my_task.get("title") );
+
+  // using an object
+my_task.set({title: "Run a hash of object"});
+  // prints "Title:  Create a model"
+console.log("Title: " + my_task.get("title") );
+  // prints "Complete:  true"
+console.log("Complete: " + my_task.get("completed") );
+```
+
+That's... great, but lets get something output to the webpage.
+
+### Creating An Underscore Template
+
+We will use an underscore template to take the data from the Model & render it like this.
+
+![Backbone Models & Underscore Templates](images/model-and-template.png)
+
+Lets use our fabulous underscore library to create a template to render, or draw our task item.  We have the following template in our HTML file to use.  Notice the `<%= %>` handles just like ERB.  We can use our underscore templates much like views in Rails to describe how we want to render our tasks.  
+
+```html
+<!-- index.html -->
+...
+  <script id="taskItemTemplate" type="text/template">
+    <li class="task-item column column-block">
+      <h2 <%= completed ? 'class="completed"' : 'class="incomplete"' %> >
+        <strong> <%= title %></strong>
+      </h2>
+      <div class="row">
+        <div class="small-6 columns">
+          <button class="button success">
+            Toggle <%= completed ? "Incomplete" : "Complete" %>
+          </button>
+        </div>
+        <div class="small-6 columns">
+          <button class="button alert">
+            Delete
+          </button>
+        </div>
+      </div>
+    </li>
+  </script>
+```
+
+Note that we can use the ternary operator `?` to change the output depending on values passed into the template.  In this case we are changing the class of the title's `<h2>` element and changing the text displayed on a button based on the `completed` property. 
+
+Next we'll use the template to add content to the DOM.  
+ 
+
+```javascript
+// src/app.js
+
+...
+
+$(document).ready(function() {
+
+  var my_task = new Task({
+    title: "Create a model",
+    completed: true
+  });
+
+  // Replacing the console log statements.
+  
+    // Select the template using jQuery
+  var template_text = $('#taskItemTemplate').html();
+  
+    // Get an underscore template object
+  var template = _.template(template_text);
+  
+    // Use the underscore template function to compile the
+    // template and data into raw html.
+  var compiledHTML = template(my_task.toJSON());
```

Comment:
It's because _ expects a simple JS object as the template parameter, not a Backbone Model.